python/TestHarness/schedulers/RunParallel.py

```diff
@@ -0,0 +1,130 @@
+from timeit import default_timer as clock
+
+from Tester import Tester
+from Scheduler import Scheduler
+from MooseObject import MooseObject
+from util import *
+import platform, time
+
+import os
+
+## This class provides an interface to run commands in parallel
+#
+# To use this class, call the .run() method with the tester options
+class RunParallel(Scheduler):
+    @staticmethod
+    def validParams():
+        params = Scheduler.validParams()
+        params.addRequiredParam('scheduler',    'RunParallel', "The name of this scheduler")
+
+        return params
+
+    def __init__(self, harness, params):
+        Scheduler.__init__(self, harness, params)
+
+    ## Run the test!
+    ## Note: use thread_lock where necessary
+    #
+    #  with thread_lock:
+    #      do protected stuff
+    #
+    # The run method should be blocking until the test has completed _and_ the results have been
+    # processed (processResults()). When we return from this method, this test will be considered
+    # finished and added to the status pool for printing out the final results.
+    def run(self, tester, testers, thread_lock, options):
+        # Ask the tester if its allowed to run on this machine
+        if tester.checkRunnableBase(options):
+
+            # Ask if this test is allowed to run based on other test's circumstances
+            status_check = StatusDependency(tester, testers, options)
+            if status_check.checkAndSetStatus():
+                # Get the command needed to run this test
+                command = tester.getCommand(options)
+
+                # Prepare to run the test
+                tester.prepare(options)
+
+                # If shouldExecute was false, there is process or output to work with so pass
+                # control over to testOutput now, to run processResults and complete the test
+                if not tester.shouldExecute() or options.dry_run:
+                    tester.start_time = clock()
+                    tester.exit_code = 0
+                    tester.end_time = clock()
+                    tester.std_out = ''
```

Comment:
Nope! You are digging into the Tester object from the Scheduler object to change it's internal state. The Tester should change it's own state.